### PR TITLE
initial enabling of centos ops-base container

### PR DESCRIPTION
**Pull request by joelddiaz**

### CODE REVIEW

**Comment by joelddiaz** [comment]
@kwoodson @twiest let me know what you think of this attempt to get centos containers built

File: docker/oso-ops-base/centos7/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
+FROM centos:centos7
```
Comment (twiest) [inline docker/oso-ops-base/centos7/Dockerfile]: Add huge banner saying this is autogenerated and any changes will get stomped on during the next generation.


```
   _____ ______ _   _ ______ _____         _______ ______ _____             _____   ____    _   _  ____ _______   ______ _____ _____ _______ 
  / ____|  ____| \ | |  ____|  __ \     /\|__   __|  ____|  __ \           |  __ \ / __ \  | \ | |/ __ \__   __| |  ____|  __ \_   _|__   __|
 | |  __| |__  |  \| | |__  | |__) |   /  \  | |  | |__  | |  | |  ______  | |  | | |  | | |  \| | |  | | | |    | |__  | |  | || |    | |   
 | | |_ |  __| | . ` |  __| |  _  /   / /\ \ | |  |  __| | |  | | |______| | |  | | |  | | | . ` | |  | | | |    |  __| | |  | || |    | |   
 | |__| | |____| |\  | |____| | \ \  / ____ \| |  | |____| |__| |          | |__| | |__| | | |\  | |__| | | |    | |____| |__| || |_   | |   
  \_____|______|_| \_|______|_|  \_\/_/    \_\_|  |______|_____/           |_____/ \____/  |_| \_|\____/  |_|    |______|_____/_____|  |_|   
                                                                                                                                            
```

File: docker/oso-ops-base/centos7/build.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# We MUST be in the same directory as this script for the build to work properly
+cd $(dirname $0)
+
+
+# Build ourselves
+echo
+echo "Building oso-centos7-ops-base..."
+sudo time docker build $@ -t oso-centos7-ops-base .
+DOCKER_EXITCODE=$?
+
```
Comment (twiest) [inline docker/oso-ops-base/centos7/build.sh]: Still need to tag it

File: docker/oso-ops-base/centos7/build.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+RED="$(echo -e '\033[1;31m')"
+NORM="$(echo -e '\033[0m')"
+
+sudo echo -e "\nTesting sudo works...\n"
+
+function is_rhel()
+{
+  grep -qi 'Red Hat Enterprise Linux' /etc/redhat-release
+  return $?
+}
```
Comment (twiest) [inline docker/oso-ops-base/centos7/build.sh]: Remove this

File: docker/oso-ops-base/centos7/push.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+
+echo
+echo "Pushing oso-centos7-ops-base..."
+sudo docker push docker-registry.ops.rhcloud.com/ops/oso-centos7-ops-base
```
Comment (twiest) [inline docker/oso-ops-base/centos7/push.sh]: We should push the centos7 container to docker hub

File: docker/oso-ops-base/rhel7/Dockerfile [hunk 5]
```diff
@@ -1,5 +1,7 @@
 FROM rhel7:latest
 
+ADD etc-pki-entitlement /etc-pki-entitlement
```
Comment (twiest) [inline docker/oso-ops-base/rhel7/Dockerfile]: This is only needed for Fedora. Build script should add it if needed. Old way used sed.

File: docker/oso-ops-base/src/Dockerfile.j2 [hunk 6]
```diff
@@ -0,0 +1,71 @@
+{% if True: %}
+#     ___ ___ _  _ ___ ___    _ _____ ___ ___         
+#    / __| __| \| | __| _ \  /_\_   _| __|   \        
+#   | (_ | _|| .` | _||   / / _ \| | | _|| |) |       
+#    \___|___|_|\_|___|_|_\/_/_\_\_|_|___|___/_ _____ 
+#   |   \ / _ \  | \| |/ _ \_   _| | __|   \_ _|_   _|
+#   | |) | (_) | | .` | (_) || |   | _|| |) | |  | |  
+#   |___/ \___/  |_|\_|\___/ |_|   |___|___/___| |_|  
+#
```
Comment (twiest) [inline docker/oso-ops-base/src/Dockerfile.j2]: Change this to a role variable and include like:

```yaml
{{ generated_header }}
```

File: docker/roles/generate_containers/README.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+------------
+
+N/A
+
+Role Variables
+--------------
+
+base_os: the type of container you want to generate (ie rhel7/centos7)
+template_list: list of files that are produced from templates (.j2 extension)
+executable_list: list of files that need to be executable for docker build
```
Comment (twiest) [inline docker/roles/generate_containers/README.md]: FIX!!!

**Comment by twiest** [comment]
:+1:

File: docker/oso-ops-base/src/Dockerfile.j2 [hunk 8]
```diff
@@ -22,9 +29,11 @@ ENV PYTHONSTARTUP=/root/.pythonrc
 # setup epel repo
 RUN rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
 
+{% if base_os == "rhel7" %}
```
Comment (kwoodson) [inline docker/oso-ops-base/src/Dockerfile.j2]: Could you combine this step with the if statement above?

**Comment by kwoodson** [comment]
:+1:

**Pull request merged**